Feat: Implement Vacation Mode #510

Geda173 opened the pull request:
## Description
This PR implements vacation mode , a game protection feature that allows players to protect their account during extended absences. When activated, players are shielded from all hostile interactions and their resource production is halted.

## Activation Requirements

- Player's own fleets must not be in transit
- Foreign fleets already en route to player's planets will complete their missions
- Minimum duration enforced: 48 hours
- No cooldown period after deactivation

## While active

1. **Outgoing Restrictions** - Players in vacation mode cannot send any fleet missions
2. **Incoming Protection** - Other players cannot send new missions targeting vacation mode players
3. **Production Halt** - All resource production set to zero, including base production (must be manually set to 100% again after leaving vacation mode)
4. **Existing Missions** - Foreign fleet missions already in transit complete normally
5. Building, Research, and Shipyard queues are paused

## Visual indicators

- Galaxy view displays vacation mode players with blue names and "(v)" marker
- Options page shows current status and deactivation eligibility date

## Mission modifications

- **Sender check**: Blocks mission dispatch if sending player is in vacation mode
- **Target check**: Blocks new missions targeting players in vacation mode (for applicable mission types)

### Type of Change:
- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Other (please describe):

## Related Issues
Fixes #510 

## Checklist
Before submitting this pull request, ensure all following requirements as outlined in [CONTRIBUTING.md](https://github.com/lanedirt/OGameX/blob/main/CONTRIBUTING.md) are met:

- [X] **Code Standards:** Code adheres to PSR-12 coding standards. Verified with Laravel Pint.
- [X] **Static Analysis:** Code passes PHPStan static code analysis.
- [X] **Testing:** Comprehensive test suite with 11 vacation mode tests covering all functionality
- [X] **CSS & JS Build:** Dialogue options for activating vacation mode were added.
- [X] **Documentation:** Inline code documentation added via docblocks and comments.

lanedirt commented:
Hi @Geda173,

Awesome work, implementation looks pretty solid! 

A usability question though: did you look into adding a confirmation dialog? As with the current implementation, once you check the checkbox and press save, vacation mode immediately starts and there is no way to cancel it except to wait for 48 hours. 

Some players might want to play around with settings and see what this option does, not knowing or expecting that it immediately activates it without an (extra) confirmation.

The current OGame does show a confirm dialog when pressing the "Activate" button:
<img width="669" height="391" alt="image" src="https://github.com/user-attachments/assets/ba14a2b0-080b-4092-9b7d-b4e4e6ef64fb" />

Second thing that caught my eye, when trying to disable vacation mode before the 48h are up: a warning message appears on screen (which is good), but also a duplicate white text string appears on top of settings header, which looks out of place:

<img width="623" height="189" alt="image" src="https://github.com/user-attachments/assets/7ccf7893-03c4-4c89-83af-fb0bec9a691f" />

Geda173 commented:
I have removed the string of text and added the dialogue prior to activating vacation mode. Another thing we could implement to make it more "sleek" is an activate/deactivate button instead of ticking a box. Your screenshot featuring the confirmation dialogue already has it in the background:

<img width="644" height="301" alt="image" src="https://github.com/user-attachments/assets/1e2ed0fe-1e04-47b8-8850-08e021cc0a3a" />

lanedirt commented:
Thanks for the fixes! I have just tested it again, and the new confirm window looks good! Thanks for adding that.

---

> I have removed the string of text

I am still seeing the duplicate white string of text with the latest version (unless it's a caching issue on my side): 
<img width="662" height="257" alt="image" src="https://github.com/user-attachments/assets/ce29f119-419f-4e33-a2a1-52aef1b98c2f" />

---

> Another thing we could implement to make it more "sleek" is an activate/deactivate button instead of ticking a box. 

Yes that would indeed be nice 👍 , making it match the current OGame style. They must have changed this since I made the copy of the settings frontend. However having it like it is currently is also fine by me, so happy to merge this as-is but if you have time to add the button logic, feel free to do so!
